jaeger/cmd/check.go

```diff
@@ -0,0 +1,189 @@
+			WithCheck(func(ctx context.Context) error {
+				// Check for Collector Service Account
+				kubeAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+				if err != nil {
+					return err
+				}
+
+				return healthcheck.CheckServiceAccounts(ctx, kubeAPI, []string{"collector"}, "linkerd-jaeger", "")
+			}))
+
+	checkers = append(checkers,
+		*healthcheck.NewChecker("jaeger service account exists", "", false, true, time.Time{}, false).
+			WithCheck(func(ctx context.Context) error {
+				// Check for Jaeger Service Account
+				kubeAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+				if err != nil {
+					return err
+				}
+
+				return healthcheck.CheckServiceAccounts(ctx, kubeAPI, []string{"jaeger"}, "linkerd-jaeger", "")
+			}))
+
+	checkers = append(checkers,
+		*healthcheck.NewChecker("collector config map exists", "", false, true, time.Time{}, false).
+			WithCheck(func(ctx context.Context) error {
+				// Check for Jaeger Service Account
+				kubeAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+				if err != nil {
+					return err
+				}
+
+				_, err = kubeAPI.CoreV1().ConfigMaps("linkerd-jaeger").Get(ctx, "collector-config", metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				return nil
+			}))
+	checkers = append(checkers,
+		*healthcheck.NewChecker("collector pod is running", "", false, true, time.Time{}, false).
+			WithCheck(func(ctx context.Context) error {
+				// Check for Collector pod
+				kubeAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+				if err != nil {
+					return err
+				}
+
+				pods, err := kubeAPI.GetPodsByNamespace(ctx, "linkerd-jaeger")
+				if err != nil {
+					return err
+				}
+
+				return healthcheck.CheckContainerRunning(pods, "collector")
+			}))
+
+	checkers = append(checkers,
+		*healthcheck.NewChecker("jaeger pod is running", "", false, true, time.Time{}, false).
+			WithCheck(func(ctx context.Context) error {
+				// Check for Jaeger pod
+				kubeAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+				if err != nil {
+					return err
+				}
+
+				pods, err := kubeAPI.GetPodsByNamespace(ctx, "linkerd-jaeger")
+				if err != nil {
+					return err
+				}
+
+				return healthcheck.CheckContainerRunning(pods, "jaeger")
+			}))
+
+	return healthcheck.NewCategory(LinkerdJaegerExtensionCheck, checkers, true)
+}
+
+func newCheckOptions() *checkOptions {
+	return &checkOptions{
+		wait:      300 * time.Second,
+		namespace: "",
+		output:    healthcheck.TableOutput,
+	}
+}
+
+// checkFlagSet specifies flags allowed with and without `config`
+func (options *checkOptions) checkFlagSet() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("check", pflag.ExitOnError)
+
+	flags.StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace to use for --proxy checks (default: all namespaces)")
```

Comment:
This should be used for data plane checks, but I don't see those implemented here.

Reply:
Removed it.

Should we not have a flag for user to be able to specify which ns to check for the jaeger extension pods checks? 🤔 Currently, Its dynamic (ie based on labels) as you suggested!